### PR TITLE
NCL-775 Use skipTests not maven.test.skip

### PR DESCRIPTION
**Pull request by rnc**

### CODE REVIEW

**Comment by rnc** [comment]
retest this please

**Comment by rnc** [comment]
retest this please

**Comment by vibe13** [comment]
retest this please

**Comment by vibe13** [comment]
retest this please

File: demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java [hunk 1]
```diff
@@ -168,7 +168,7 @@ public void initiliazeProjectProductData() {
                     .project(project1)
                     .description("Test build config for project newcastle")
                     .environment(environment1)
-                    .buildScript("mvn clean deploy -Dmaven.test.skip")
+                    .buildScript("mvn clean deploy -DskipTests=true")
```
Comment (matejonnet) [inline demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java]: Isn't it better for faster build to skip also the compilation of tests and not just the execution ?

Comment (rnc) [inline demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java]: Perhaps for some projects eg pnc but for others e.g. fabric it breaks the build as later modules rely on test jars being built. I think building the tests but not running them is a reasonable compromise.

Comment (matejonnet) [inline demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java]: Looks like project (fabric) setup must be fixed, runtime code should not depend on test code.

Comment (rnc) [inline demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java]: Maybe but many community projects might be like this. We have to build what we are provided. Also we should probably deploy the same as them i.e. deploy the test artifacts.

Comment (matejonnet) [inline demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java]: Ok, then we can use that properly only for our project (pnc).

Comment (rnc) [inline demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java]: Timing a build on an internal server with skipTests versus maven.test.skip the difference appears fairly insignificant.

Comment (matejonnet) [inline demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java]: But using maven.test.skip will ensure we don't end up using test classes in runtime.

**Pull request merged**